### PR TITLE
Fix PDF date range bug

### PR DESCRIPTION
**Pull request by conbrad**
- Iterate over dates in selected prep range, not just available dailies
- Functionally correct output, although table is still missing row cells for missing daily data, but perfect is the enemy of good.
# Test Links:
[Percentile Calculator](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Fire Behaviour Advisory](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2028.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

**Review by andrea-williams — APPROVED** [review]

**Review by tessmgood — APPROVED** [review]
fix looks good to me - i'm going to find out from users if they'd like to see some ND cues in the table (this was discussed previously). ND = no data, the convention they used in HFI workbooks. At first glance the export seems broken, if you look at them side by side you realize what's going on (time consuming), in the future if someone is looking at this to understand decisions made / for legal reasons - the output will appear broken / may not be trustworthy/credible. 
TL;DR - good fix, we likely need to do more though, given the emergency context and record keeping requirements
